mitodl/ccxcon
# Implemented filtering out hidden chapters

#### What are the relevant tickets?

Fixes #102 
#### What's this PR do?

It passes the new `visible_to_staff_only` field to edX when requesting modules so we can get the hidden status, then it filters the hidden chapters out.
#### Where should the reviewer start?

courses/tasks.py
#### How should this be manually tested?

First, make sure you have CCXCon properly linked to your devstack or some other instance of openedX.
- if the PR is not yet merged into master check out the branch for https://github.com/mitocw/edx-platform/pull/210
- In CMS/studio go to the edX demo course. Find a chapter and click the gear settings icon next to it. Check 'Hide from students' and save.
- Then go to advanced settings, adjust some setting and save. This will cause the edX instance to POST the course to CCXCon, which will cause CCXCon to attempt to retrieve the module information from edX.
- On CCXCon look at `/api/v1/coursexs/` and click on the modules link for the course you updated. The module you marked as hidden should no longer be there, but all other modules should be present.
#### What GIF best describes this PR or how it makes you feel?

![](http://i.imgur.com/cHJzdJc.gif)
